src/object/cli_shard.c

```diff
@@ -154,22 +154,25 @@ dc_rw_cb(tse_task_t *task, void *arg)
 		 * If any failure happens inside Cart, let's reset failure to
 		 * TIMEDOUT, so the upper layer can retry.
 		 */
-		D_ERROR("RPC %d failed: %d\n",
-			opc_get(rw_args->rpc->cr_opc), ret);
+		D_ERROR("RPC %d failed: %d\n", opc, ret);
 		D_GOTO(out, ret);
 	}
 
 	rc = obj_reply_get_status(rw_args->rpc);
 	if (rc != 0) {
-		D_ERROR("rpc %p RPC %d failed: %d\n", rw_args->rpc,
-			 opc_get(rw_args->rpc->cr_opc), rc);
+		if (rc == -DER_INPROGRESS)
+			D_DEBUG(DB_TRACE, "rpc %p RPC %d may need retry: %d\n",
```

Comment:
should we use DB_IO?   Or is it too chatty?   The issue I have with TRACE is that you won't see these messages without turning on tons of other ones.